src/operator/nn/mkldnn/mkldnn_act.cc

```diff
@@ -147,6 +147,8 @@ static MKLDNNActForward &GetActForward(const ActivationParam& param,
   auto it = fwds.find(key);
   if (it == fwds.end()) {
     MKLDNNActForward fwd(param, ctx.is_train, in_data, in_mem);
+    if (MKLDNNCacheExceeded(fwds.size()))
+      fwds.clear();
```

Comment:
Shall we just randomly remove one from fwds and insert the new one? It's more reasonable to remove the first inserting one that not be reused, but it's hard to achieve. Randomly removing is acceptable. Our target is balancing cache size with performance. Flushing cache isn't the best choice.